source/gui/guiHelper.py

```diff
@@ -154,6 +154,52 @@ def associateElements( firstElement, secondElement):
 
 	return sizer
 
+
+class BaseLabeledControl:
+	"""Base class used to create a labeled control subclass
+	A labeled control subclass allows to create a control with an associated label.
+	The associated label is enabled and disabled when the main control is enabled or disabled (respectively).
+	
+	Example usage:
+	
+	class LabeledChoice(BaseLabeledControl, wx.Choice):
+		pass
+	myChoice = LabeledChoice(
+		parent,
+		labelText='Choose an option',
+		wxCtrlClass=wx.Choice,
+		choices=['Option 1', 'Option 2', 'Option 3'],
+	)
+	"""
+
+	def __init__(self, parent, labelText, wxCtrlClass, **kwargs):
+		""" @param parent: An instance of the parent wx window. EG wx.Dialog
+			@param labelText: The text to associate with a wx control.
+			@type labelText: string
+			@param wxCtrlClass: The class to associate with the label, eg: wx.TextCtrl
+			@type wxCtrlClass: class
+			@param kwargs: The keyword arguments used to instantiate the wxCtrlClass
+		"""
+		self.label = wx.StaticText(parent, label=labelText)
+		if self.__class__ == BaseLabeledControl:
+			raise TypeError(
+				f"BaseLabeledControl should be subclassed and should also inherit from {wxCtrlClass.__name__}."
+			)
+		# Check that the final class also inherits from wxCtrlClass
+		if not isinstance(self, wxCtrlClass):
+			raise TypeError(f"{self.__class__.__name__} should be a subclass of {wxCtrlClass.__name__}.")
+		self.controlClass = wxCtrlClass
+		self.controlClass.__init__(self, parent, **kwargs)

```

Comment:
I don't think this is quite right. I think it means that `__new__` won't get called on `wxCtrlClass` which could result in hard to diagnose issues.